### PR TITLE
TAV-415: Add db level repo tests.

### PR DESCRIPTION
**Pull request by ssaleem-ee**

### CODE REVIEW

File: test/it/it/suites/DbBaseSpec.scala [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      user = container.username,
+      pass = container.password
+    )
+
+//    val ptx = Transactor.fromDriverManager[IO](
+//      driver = "org.postgresql.Driver",
+//      url = "jdbc:postgresql://localhost:5433/addressbasepremium",
+//      user = "root",
+//      pass = "password"
+//    )
```
Comment (Ardesco) [inline test/it/it/suites/DbBaseSpec.scala:70]: Can this comment be removed?  Looks obsolete.

File: test/it/it/suites/DbBaseSpec.scala [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      _ <- csql(s"""CREATE SCHEMA ${schemaName}""").update.run
+      _ =  println(s""">>> Created schemaName: ${schemaName}""")
+      _ <- csql(sqlMap("createDbSchemaUrl")).update.run
+      _ =  println(s""">>> Created schema objects""")
+      _ <- csql("""CREATE TABLE IF NOT EXISTS public.address_lookup_status (
+                   | schema_name VARCHAR(64) NOT NULL PRIMARY KEY,
+                   | status      VARCHAR(32) NOT NULL,
+                   | error_message VARCHAR NULL,
+                   | timestamp   TIMESTAMP NOT NULL);""".stripMargin).update.run
+      _ =  println(s""">>> Created public.address_lookup_status table""")
```
Comment (Ardesco) [inline test/it/it/suites/DbBaseSpec.scala:88]: Have we got any check to ensure that this structure matches the real DB structure?

**Comment by Ardesco** [comment]
Is the instance of Address Lookup spun up by service manager going to be expecting a postgres DB to be available now?  If so I don't think we have all the postcodes used by the acceptance tests:

https://github.com/hmrc/address-lookup-acceptance-tests/blob/master/src/test/scala/uk/gov/hmrc/integration/testdata/Addresses.scala

Secondly, will this need an updated sm config to be able to spin up a postgres instance?

I'm unclear how this is going to fit together for the  acceptance test run on jenkins.

File: test/it/it/suites/DbBaseSpec.scala (hunk 1 above)
Comment (ssaleem-ee) [reply test/it/it/suites/DbBaseSpec.scala:70]: Yep - in there so that I can debug locally.

File: test/it/it/suites/DbBaseSpec.scala (hunk 2 above)
Comment (ssaleem-ee) [reply test/it/it/suites/DbBaseSpec.scala:88]: The structure for the tables and views other than address_lookup_status is pulled directly from the lambda repo. These tests essentially check that the AddressSearchRepo works with the db schema as expected.

File: test/it/it/suites/DbBaseSpec.scala (hunk 1 above)
Comment (Ardesco) [reply test/it/it/suites/DbBaseSpec.scala:70]: Can't we pass in `containt.*` properties on the command line to set the URL, user and pass?

**Comment by ssaleem-ee** [comment]
This is a thin slice of the testing stack and is meant to plug a specific hole that the existing tests don't cover - so no need to run this as part of the acceptance tests. The only thing that must be available and running for the test to run is docker - there are no other dependencies that are not pulled in as per normal.

File: test/it/it/suites/DbBaseSpec.scala (hunk 2 above)
Comment (ssaleem-ee) [reply test/it/it/suites/DbBaseSpec.scala:88]: There is another bit of work that could/should be done and that is to pull out all the sql into files in the ingest lambda repo. Then, this test will reference those directly and we won't need to inline the sql here + nothing to keep in sync.

File: test/it/it/suites/DbBaseSpec.scala (hunk 1 above)
Comment (ssaleem-ee) [reply test/it/it/suites/DbBaseSpec.scala:70]: We can but then we run the risk of port clashes. This way, TestContainers picks a port randomly and we don't need to care.

**Review by Ardesco — APPROVED** [review]

**Review by Ardesco — APPROVED** [review]

**Pull request merged**